blockindex/contractstaking/indexer.go

```diff
@@ -121,9 +121,16 @@ func (s *Indexer) BucketTypes() ([]*BucketType, error) {
 
 // PutBlock puts a block into indexer
 func (s *Indexer) PutBlock(ctx context.Context, blk *block.Block) error {
-	if blk.Height() < s.contractDeployHeight || blk.Height() <= s.cache.Height() {
+	expectHeight := s.cache.Height() + 1
+	if expectHeight < s.contractDeployHeight {
+		expectHeight = s.contractDeployHeight
+	}
+	if blk.Height() < expectHeight {
```

Comment:
let's still use
```
if blk.Height() < s.contractDeployHeight {
    return nil
}
expectHeight := s.cache.Height() + 1
...
```
i think this is more clear and easy to understand

Reply:
`expectHeight` will be 1 when `blk.Height() == s.contractDeployHeight` if in this way